ioctl/newcmd/action/actionhash.go

```diff
@@ -0,0 +1,319 @@
+			result += fmt.Sprintf("  payload: %s\n", transfer.Payload)
+		}
+		result += ">\n"
+	case core.GetExecution() != nil:
+		execution := core.GetExecution()
+		result += "execution: <\n" +
+			fmt.Sprintf("  contract: %s %s\n", execution.Contract,
+				Match(execution.Contract, "address"))
+		if execution.Amount != "0" {
+			amount, err := util.StringToIOTX(execution.Amount)
+			if err != nil {
+				return "", errors.New("failed to convert string into IOTX amount")
+			}
+			result += fmt.Sprintf("  amount: %s IOTX\n", amount)
+		}
+		result += fmt.Sprintf("  data: %x\n", execution.Data) + ">\n"
+	case core.GetPutPollResult() != nil:
+		putPollResult := core.GetPutPollResult()
+		result += "putPollResult: <\n" +
+			fmt.Sprintf("  height: %d\n", putPollResult.Height) +
+			"  candidates: <\n"
+		for _, candidate := range putPollResult.Candidates.Candidates {
+			result += "    candidate: <\n" +
+				fmt.Sprintf("      address: %s\n", candidate.Address)
+			votes := big.NewInt(0).SetBytes(candidate.Votes)
+			result += fmt.Sprintf("      votes: %s\n", votes.String()) +
+				fmt.Sprintf("      rewardAdress: %s\n", candidate.RewardAddress) +
+				"    >\n"
+		}
+		result += "  >\n" +
+			">\n"
+	default:
+		result += protoV1.MarshalTextString(core)
+	}
+	result += fmt.Sprintf("senderPubKey: %x\n", action.SenderPubKey) +
+		fmt.Sprintf("signature: %x\n", action.Signature)
+
+	return result, nil
+}
+
+func printReceiptProto(receipt *iotextypes.Receipt) string {
+	result := fmt.Sprintf("status: %d %s\n", receipt.Status,
+		Match(strconv.Itoa(int(receipt.Status)), "status")) +
+		fmt.Sprintf("actHash: %x\n", receipt.ActHash) +
+		fmt.Sprintf("blkHeight: %d\n", receipt.BlkHeight) +
+		fmt.Sprintf("gasConsumed: %d\n", receipt.GasConsumed) +
+		printLogs(receipt.Logs)
+	if len(receipt.ContractAddress) != 0 {
+		result += fmt.Sprintf("\ncontractAddress: %s %s", receipt.ContractAddress,
+			Match(receipt.ContractAddress, "address"))
+	}
+	if len(receipt.Logs) > 0 {
+		if index, ok := staking.BucketIndexFromReceiptLog(receipt.Logs[0]); ok {
+			result += fmt.Sprintf("\nbucket index: %d", index)
+		}
+	}
+	if receipt.Status == uint64(iotextypes.ReceiptStatus_ErrExecutionReverted) {
+		result += fmt.Sprintf("\nexecution revert reason: %s", receipt.ExecutionRevertMsg)
+	}
+	return result
+}
+
+func printLogs(logs []*iotextypes.Log) string {
+	result := "logs:<\n"
+	for _, l := range logs {
+		result += "  <\n" +
+			fmt.Sprintf("    contractAddress: %s\n", l.ContractAddress) +
+			"    topics:<\n"
+		for _, topic := range l.Topics {
+			result += fmt.Sprintf("      %s\n", hex.EncodeToString(topic))
+		}
+		result += "    >\n"
+		if len(l.Data) > 0 {
+			result += fmt.Sprintf("    data: %s\n", hex.EncodeToString(l.Data))
+		}
+		result += "  >\n"
+
+	}
+	result += ">\n"
+	return result
+}
+
+// Match returns human readable expression
+func Match(in string, matchType string) string {
+	switch matchType {
+	case "address":
+		alias, err := alias.Alias(in)
```

Comment:
don't use ioctl/cmd/xxx,   add function in client interface to implement, then call it like `client.XXX`